revolutionpi/gpio_chip.go

```diff
@@ -135,14 +161,14 @@ func (g *gpioChip) Close() error {
 	return err
 }
 
-func (g *gpioChip) findDIODevice(gpioAddress uint16) (SDeviceInfo, error) {
-	for _, dev := range g.dioDevices {
+func findDevice(address uint16, deviceList []SDeviceInfo) (SDeviceInfo, error) {
```

Comment:
renamed and passed in device info in order to have less duplicated code, though I am passing in an array of structs and I don't know if that is worse